compiler/rustc_resolve/src/lib.rs

```diff
@@ -1230,6 +1231,7 @@ impl<'tcx> Resolver<'_, 'tcx> {
 
         let feed = self.tcx.feed_local_def_id(def_id);
         feed.def_kind(def_kind);
+        feed.def_span(rustc_middle::util::lower_span(self.tcx, def_span, def_id));
```

Comment:
I fear we have subtlety here around relative spans. Hashing relative spans needs to check `source_span`, which is set a few lines below.
Can you try with the `source_span` lines moves between `create_def` and `feed_local_def_id`?
We may need to make `source_span` more robust.

Reply:
I'm not certain if it makes sense because one of is `﻿span` and the other is ﻿`def_span`, which are different instances.

Reply:
Calling `feed.def_span` on `def_span.with_parent(def_id)` will attempt to hash `def_span.with_parent(def_id)`.
As `def_span.with_parent(def_id)` is relative to `def_id`, this will look at `source_span[def_id]` as container span.
If we have not yet assigned to `source_span[def_id]`, `def_span.with_parent(def_id)` will be hashed as an absolute span, while it should be hashed as relative to `source_span[def_id]`.